test/unittests/libevm/VMTest.cpp

```diff
@@ -577,6 +577,84 @@ class AlethInterpreterChainIDTestFixture : public ChainIDTestFixture
     {}
 };
 
+class SelfBalanceFixture : public TestOutputHelperFixture
+{
+public:
+    explicit SelfBalanceFixture(VMFace* _vm) : vm{_vm} { state.addBalance(address, 1 * ether); }
+
+    void testSelfBalanceWorksInIstanbul()
+    {
+        ExtVM extVm(state, envInfo, *se, address, address, address, value, gasPrice, {}, ref(code),
+            sha3(code), version, depth, isCreate, staticCall);
+
+        owning_bytes_ref ret = vm->exec(gas, extVm, OnOpFunc{});
+
+        BOOST_REQUIRE_EQUAL(fromBigEndian<u256>(ret), 1 * ether);
+    }
+
+    void testSelfBalanceHasCorrectCost()
+    {
+        ExtVM extVm(state, envInfo, *se, address, address, address, value, gasPrice, {}, ref(code),
+            sha3(code), version, depth, isCreate, staticCall);
+
+        bigint gasBefore;
+        bigint gasAfter;
+        auto onOp = [&gasBefore, &gasAfter](uint64_t /*steps*/, uint64_t /* PC */,
+                        Instruction _instr, bigint /*newMemSize*/, bigint /*gasCost*/, bigint _gas,
+                        VMFace const*, ExtVMFace const*) {
+            if (_instr == Instruction::SELFBALANCE)
+                gasBefore = _gas;
+            else if (gasBefore != 0 && gasAfter == 0)
+                gasAfter = _gas;
+        };
+
+        vm->exec(gas, extVm, onOp);
+
+        BOOST_REQUIRE_EQUAL(gasBefore - gasAfter, 5);
+    }
+
+    void testSelfBalanceisInvalidBeforeIstanbul()
+    {
+        se.reset(ChainParams(genesisInfo(Network::ConstantinopleFixTest)).createSealEngine());
+        version = ConstantinopleFixSchedule.accountVersion;
```

Comment:
How come the account version is available in the Constantinople Fix schedule..do we need to reference it in this test since from what I understand it's been punted to post-Istanbul?

Reply:
Versioning still exists in the code for all forks, with version being 0 in all forks before we activate version 1 (currently it's `Experimental` fork)

This line is actually redundant.